src/BarChart/BarSeries/BarTargetMarker.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { FC, useMemo } from 'react';
+import { motion } from 'framer-motion';
+import {
+  ChartInternalShallowDataShape,
+  DEFAULT_TRANSITION,
+  Direction
+} from '@/common';
+
+export interface BarTargetMarkerProps {
+  /**
+   * Height of the bar.
+   */
+  height: number;
+
+  /**
+   * Width of the bar.
+   */
+  width: number;
+
+  /**
+   * SVG x attribute for the bar.
+   */
+  x: number;
+
+  /**
+   * SVG y attribute for the bar.
+   */
+  y: number;
+
+  /**
+   * Group index or index of the bar. Set internally by `BarSeries`.
+   */
+  index: number;
+
+  /**
+   * D3 scale for Axis. Set internally by `BarChart`.
+   */
+  scale: any;
+
+  /**
+   * Parsed data shape. Set internally by `BarChart`.
+   */
+  data: ChartInternalShallowDataShape;
+
+  /**
+   * Color of the target marker line.
+   */
+  fill: string;
+
+  /**
+   * Color of the positive delta line.
+   */
+  positiveDeltaFill: string;
```

Comment:
I'm wondering if these colors should follow the same standard color interface as the rest.

Reply:
We don't need color schema here, for simple elements like label, etc we used just string

Reply:
The interface supports callbacks and whatnot is what i meant

Reply:
the utils which we have for getting color from color schema look like that. So I need to pass all that info for getting color even if it just a string
```
      const fill = getColor({
        data,
        colorScheme,
        point: item.data,
        index
      });
```

Reply:
<img width="512" alt="Screenshot 2024-11-21 at 15 11 14" src="https://github.com/user-attachments/assets/067cbf54-7390-4d70-b1db-052d86fb3ca2">

As I understood we are using `ColorSchemeType` for Series data, for other components it's just `string`

<img width="503" alt="Screenshot 2024-11-21 at 15 13 16" src="https://github.com/user-attachments/assets/cf89a801-bf66-40c2-b6b7-845b34722201">

Reply:
Ah, I didn't realize it was coupled to color scheme so much. Fine for now.